### PR TITLE
Allow dogstatsd sidecar

### PR DESCRIPTION
**Pull request by jesusfcr**
Added dogstatsd sidecar to following compoments:

- Not infra components or insights (just proxy).
- Available in localhost: 8125
- Enabled by default only in metrics
- Enable/disable with  <component>.dogstatsd.enabled=truie

Migration in gitops - add apiKey:

```
dogstatsd:
  apiKey: {{ fetch "secrets:/vulcan/k8s/metrics/dogstatsd/api/key" }}
```

### CODE REVIEW

File: stable/vulcan/templates/_dogstatsd.tpl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{- define "dogstatsd-sidecar" -}}
+{{- if .Values.dogstatsd.enabled -}}
```
Comment (danfaizer) [inline stable/vulcan/templates/_dogstatsd.tpl:2]: Is possible to add this env var:
`DD_HOSTNAME` and set default value to `dogstatsd` ?

File: stable/vulcan/templates/_dogstatsd.tpl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+- name: dogstatsd
+  image: datadog/docker-dd-agent:latest-dogstatsd
```
Comment (danfaizer) [inline stable/vulcan/templates/_dogstatsd.tpl:4]: I'm not sure which is the difference, but it looks more "natural" for me pick the standalone dogstatsd docker image:
https://hub.docker.com/r/datadog/dogstatsd
And I would pick fixed latest version: `datadog/dogstatsd:7.19.2`
WDYT?

File: stable/vulcan/templates/_dogstatsd.tpl (hunk 1 above)
Comment (jesusfcr) [reply stable/vulcan/templates/_dogstatsd.tpl:2]: done

File: stable/vulcan/templates/_dogstatsd.tpl (hunk 2 above)
Comment (jesusfcr) [reply stable/vulcan/templates/_dogstatsd.tpl]: Pinned

**Pull request merged**